actionpack/lib/action_controller/metal/conditional_get.rb

```diff
@@ -77,18 +77,16 @@ def etag(&etagger)
     #
     #   before_action { fresh_when @article, template: 'widgets/show' }
     #
-    def fresh_when(record_or_options, additional_options = {})
-      if record_or_options.is_a? Hash
-        options = record_or_options
-        options.assert_valid_keys(:etag, :last_modified, :public, :template)
-      else
-        record  = record_or_options
-        options = { etag: record, last_modified: record.try(:updated_at) }.merge!(additional_options)
+    def fresh_when(record = nil, etag: record, last_modified: nil, public: false, template: nil)
+      last_modified ||= record.updated_at if record
+
+      if etag || template
+        response.etag = combine_etags(etag: etag, last_modified: last_modified,
+          public: public, template: template)
       end
 
-      response.etag          = combine_etags(options)   if options[:etag] || options[:template]
-      response.last_modified = options[:last_modified]  if options[:last_modified]
-      response.cache_control[:public] = true            if options[:public]
+      response.last_modified = last_modified if last_modified
+      response.cache_control[:public] = true if public
```

Comment:
It's okay when it's only within the method. Besides we can cross that bridge when `stale?` or `fresh_when` needs to define something publicly within itself (which I can't imagine).